regression/getCoverage.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+# Get a unique number to prevent collision of output files
+starttime=`date +%s`
+
+# Check that the previous command succeded, if not exit.
+commandStatus()
+{
+    if [ $? -ne 0 ]; then
+	printf "[ERROR]\n"
+	echo "ERROR: See cbmcCoverage$starttime.out for more information"
+	exit 1
+    fi
+    printf "[OK]\n"
+}
+
+# Check that lcov has been installed
+printf "INFO: Checking lcov is installed "
+lcov -version > cbmcCoverage$starttime.out 2>&1
+commandStatus
+
+# Remove any previous build that may not have coverage in it.
+printf "INFO: Cleaning CBMC build "
+make clean -C ../src >> cbmcCoverage$starttime.out 2>&1
+commandStatus
+
+printf "INFO: Building CBMC with Code Coverage enabled "
+# Run the usual make target with --coverage to add gcov instrumentation
+make CXXFLAGS="--coverage" LINKFLAGS="--coverage" -C ../src >> cbmcCoverage$starttime.out 2>&1
```

Comment:
Good question. I have tried this using the compiler on my Mac which is apple's version of clang-800.0.42.1:

> _bash-3.2$ gcc --version
> Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
> Apple LLVM version 8.0.0 (clang-800.0.42.1)
> Target: x86_64-apple-darwin16.3.0
> Thread model: posix
> InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
> bash-3.2$ 

I successfully got coverage results with this version. Does this answer your question?